### PR TITLE
Create starter for cloudfoundry

### PR DESCRIPTION
**Pull request by eddumelendez**
Currently, `GcpCloudFoundryEnvironmentPostProcessor` is provided at
`spring-cloud-gcp-autoconfigre` even when application is not running
in CF. This commit introduces a new starter for that case.

Fixes gh-1854

### CODE REVIEW

File: docs/src/main/asciidoc/cloudfoundry.adoc [hunk 1]
```diff
@@ -3,6 +3,16 @@
 Spring Cloud GCP provides support for Cloud Foundry's https://docs.pivotal.io/partners/gcp-sb/index.html[GCP Service Broker].
 Our Pub/Sub, Cloud Spanner, Storage, Stackdriver Trace and Cloud SQL MySQL and PostgreSQL starters are Cloud Foundry aware and retrieve properties like project ID, credentials, etc., that are used in auto configuration from the Cloud Foundry environment.
 
+In order to take advantage of cloudfoundry make sure the following dependency is added:
```
Comment (meltsufin) [inline docs/src/main/asciidoc/cloudfoundry.adoc]:
```suggestion
In order to take advantage of the Cloud Foundry support make sure the following dependency is added:
```

File: spring-cloud-gcp-cloudfoundry/pom.xml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>spring-cloud-gcp</artifactId>
+        <groupId>org.springframework.cloud</groupId>
+        <version>1.2.0.BUILD-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>spring-cloud-gcp-cloudfoundry</artifactId>
+    <name>Spring Cloud GCP CloudFoundry</name>
```
Comment (meltsufin) [inline spring-cloud-gcp-cloudfoundry/pom.xml]:
```suggestion
    <name>Spring Cloud GCP Cloud Foundry</name>
```

File: spring-cloud-gcp-cloudfoundry/pom.xml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    <description>Spring Cloud GCP CloudFoundryModule Module</description>
```
Comment (meltsufin) [inline spring-cloud-gcp-cloudfoundry/pom.xml]:
```suggestion
    <description>Spring Cloud GCP Cloud Foundry Module</description>
```

File: spring-cloud-gcp-cloudfoundry/pom.xml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+    <dependencies>
+        <dependency>
+            <groupId>io.pivotal.cfenv</groupId>
+            <artifactId>java-cfenv</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.springframework.cloud</groupId>
+            <artifactId>spring-cloud-gcp-autoconfigure</artifactId>
```
Comment (meltsufin) [inline spring-cloud-gcp-cloudfoundry/pom.xml]: Shouldn't this go in the starter instead?

File: spring-cloud-gcp-starters/spring-cloud-gcp-starter-cloudfoundry/pom.xml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+		xsi:schemaLocation="https://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<parent>
+		<artifactId>spring-cloud-gcp-starters</artifactId>
+		<groupId>org.springframework.cloud</groupId>
+		<version>1.2.0.BUILD-SNAPSHOT</version>
+	</parent>
+	<modelVersion>4.0.0</modelVersion>
+
+	<artifactId>spring-cloud-gcp-starter-cloudfoundry</artifactId>
+	<name>Spring Cloud GCP Cloud CloudFoundry Starter</name>
```
Comment (meltsufin) [inline spring-cloud-gcp-starters/spring-cloud-gcp-starter-cloudfoundry/pom.xml]:
```suggestion
	<name>Spring Cloud GCP Cloud Cloud Foundry Starter</name>
```

**Comment by eddumelendez** [comment]
thanks to submit the commit @meltsufin 

this change make sense to me 👍

**Pull request merged**